runtime/onert/backend/train/ops/PoolLayer.cc

```diff
@@ -58,8 +58,10 @@ class MaxPool2D final : public TrainingKernelRegistry
       _op_params.stride_width = strideWidth;
       _op_params.filter_height = kernelHeight;
       _op_params.filter_width = kernelWidth;
-      _op_params.padding_values.height = (int8_t)paddingTop;
-      _op_params.padding_values.width = (int8_t)paddingLeft;
+      assert(paddingTop < (1 << 8));
+      assert(paddingLeft < (1 << 8));
+      _op_params.padding_values.height = static_cast<uint8_t>(paddingTop);
+      _op_params.padding_values.width = static_cast<uint8_t>(paddingLeft);
```

Comment:
But still, IIUC, this like might be changed like this : 

```suggestion
      _op_params.padding_values.height = static_cast<int16_t>(paddingTop);
      _op_params.padding_values.width = static_cast<int16_t>(paddingLeft);
```

Reply:
It may generate negative values. If you would like to use to `static_cast<int16_t>`, could you modify the assertions like below?
```suggestion
      assert(paddingTop < 0x8000);
      assert(paddingLeft < 0x8000);
      _op_params.padding_values.height = static_cast<int16_t>(paddingTop);
      _op_params.padding_values.width = static_cast<int16_t>(paddingLeft);
```

Reply:
Your suggetion looks good for me. I'd like to follow your suggestion and fix both maxpool and averagepool code. 

@icodo98 Could you share your opinion about this? If you also agreed with https://github.com/Samsung/ONE/pull/14166#discussion_r1789669670, I'll make an PR to update both maxpool and average pool. Or If you're still interested in this, I guess you could update this PR a bit to include a above suggestion.

Reply:
> @icodo98 Could you share your opinion about this?

Sorry for late reply, I also agree with https://github.com/Samsung/ONE/pull/14166#discussion_r1789669670.
I'll update this PR.